internal/cache/cache.go

```diff
@@ -24,20 +24,20 @@ type LoadingCache struct {
 	onEvicted  func(key string, value interface{})
 
 	sync.Mutex
-	data map[string]*cacheItem
+	data      map[string]*list.Element
+	evictList *list.List
```

Comment:
We have "natural" sorting for both LRU and LRC you mentioned already in place.

`*list.Element` is used because Go lacks "generics", I'll need to copy-paste list implementation for our type to change it's values types. Do you think we should do that?

Reply:
no, my suggestion was to keep a sorted list for evicted keys only and leave items alone. I.e. data will be map of naked interface{} (as before) and evictList will hold keys only